### PR TITLE
Added support for ResponseStatusException

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixces #275

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: problem-spring-web/src/main/java/org/zalando/problem/spring/web/advice/general/ResponseStatusAdviceTrait.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.zalando.problem.spring.web.advice.general;
+
+import org.apiguardian.api.API;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.server.ResponseStatusException;
```
Comment (cbornet) [inline problem-spring-web/src/main/java/org/zalando/problem/spring/web/advice/general/ResponseStatusAdviceTrait.java]: I think the class doesn't exist in Spring 4

Comment (whiskeysierra) [reply problem-spring-web/src/main/java/org/zalando/problem/spring/web/advice/general/ResponseStatusAdviceTrait.java]: Looks like a good opportunity to drop spring 4 support.

File: problem-spring-web/src/main/java/org/zalando/problem/spring/web/advice/general/ResponseStatusAdviceTrait.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+@API(status = EXPERIMENTAL)
+public interface ResponseStatusAdviceTrait extends AdviceTrait {
+
+    @API(status = INTERNAL)
+    @ExceptionHandler
+    default ResponseEntity<Problem> handleResponseStatusException(
+            final ResponseStatusException exception,
+            final NativeWebRequest request) {
+        return create(new HttpStatusAdapter(exception.getStatus()),exception, request);
```
Comment (lukasniemeier-zalando) [inline problem-spring-web/src/main/java/org/zalando/problem/spring/web/advice/general/ResponseStatusAdviceTrait.java]: formatting

**Comment by lukasniemeier-zalando** [comment]
👍

**Comment by lukasniemeier-zalando** [comment]
Is dropping support for Spring 4 considered to be a "breaking change" ?

**Pull request merged**